stackslib/src/chainstate/nakamoto/staging_blocks.rs

```diff
@@ -302,6 +302,44 @@ impl<'a> NakamotoStagingBlocksConnRef<'a> {
             .optional()?)
     }
 
+    /// Get the block ID of a staging block from block hash.
+    /// Used for downloads
+    pub fn get_block_id_by_block_hash(
```

Comment:
Flagging for your attention -- this is fine for Nakamoto, but not fine for Stacks 2.x.

In Stacks 2.x, the same block can be mined by two different sortitions, which means there isn't a 1-to-1 correspondence between block hashes and block IDs.  For example, the same miner can win two separate sortitions in bitcoin blocks N and N+1, and mine two identical siblings.  But, because they were mined in two different sortitions, they'd have different consensus hashes, and thus block IDs.

In Nakamoto, this is safe because the consensus hash is part of the block header, so each block hash is guaranteed to be unique.